src/main/python/rlbot/utils/virtual_environment_management.py

```diff
@@ -12,10 +12,41 @@
 class EnvBuilderWithRequirements(EnvBuilder):
 
     def __init__(self, bundle: RunnableConfigBundle, do_post_setup: bool=True):
-        super().__init__(system_site_packages=True, clear=False, with_pip=False)
+        # Install pip by default and add RLBot's already installed packages as a base set of dependencies
+        # Allows bots with requirements like "rlbot" to not have to reinstall the rlbot package
+        # Also allows bots to pin to specific versions of a package, even if a different version is installed in rlbot
+        super().__init__(system_site_packages=True, with_pip=True) 
         self.bundle = bundle
         self.do_post_setup = do_post_setup
 
+    def create(self, env_dir):
+        """
+        Create a virtual environment in a directory.
+        Changed so system site packages is configured before post_setup is ran.
+
+        :param env_dir: The target directory to create an environment in.
+        """
+
+        env_dir = os.path.abspath(env_dir)
+        context = self.ensure_directories(env_dir)
+        # See issue 24875. We need system_site_packages to be False
+        # until after pip is installed.
```

Comment:
Interesting! Looks like the referenced issue is https://bugs.python.org/issue24875. It's old and allegedly resolved, do we still need this code?

Reply:
That comment is actually from the original code in Python, I just copy+pasted the function and made post_setup() get called last. https://github.com/python/cpython/blob/3.7/Lib/venv/__init__.py#L52-L76